crates/viewer/re_renderer/src/video/mod.rs

```diff
@@ -47,17 +47,15 @@ pub enum DecodingError {
     NegativeTimestamp,
 }
 
+pub type FrameDecodingResult = Result<DecodedFrame, DecodingError>;
+
 /// Information about the status of a frame decoding.
-pub enum FrameDecodingResult {
+pub enum DecodedFrame {
```

Comment:
nit: The frame may not be decoded yet, so I'm not sure about the naming here. Maybe `DecoderOutput`? We could also just use `Result<Poll<GpuTexture2D>, DecodingError>`

Reply:
`Poll<GpuTexture2D>` would mean that the other side has to store the last known good texture and deal with all of that, also incompatible with the idea of small previews. But I like the naming suggestion!

Reply:
oh I already used up `DecoderOutput` now as an internal structure :thinking:
Hmm I can at least offer a counter argument that even on `Pending` we still tend to return a decoded frame...